src/gzip/header.cr

```diff
@@ -44,6 +44,8 @@ class Gzip::Header
       xlen = io.read_byte.not_nil!
       @extra = Bytes.new(xlen)
       io.read_fully(@extra)
+    else
+      @extra = Bytes.empty
```

Comment:
This line is not needed, `@extra` should always be initialized. Right now it doesn't work because of #3988